frontend/src/utils/createReducer.ts

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2023 Marimo. All rights reserved. */
```

Comment:
this just reduces the boiler plate when creating reducers and paring the actions/reducers. it will also auto-infer many of the types so they dont need to be re-declared